src/test/java/org/folio/circulation/domain/OverduePeriodCalculatorServiceTest.java

```diff
@@ -0,0 +1,534 @@
+package org.folio.circulation.domain;
+
+import api.support.builders.LoanBuilder;
+import api.support.builders.LoanPolicyBuilder;
+import api.support.builders.OpeningPeriodsBuilder;
+import api.support.builders.OverdueFinePolicyBuilder;
+import io.vertx.core.json.JsonObject;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.http.entity.ContentType;
+import org.folio.circulation.domain.policy.LoanPolicy;
+import org.folio.circulation.domain.policy.OverdueFinePolicy;
+import org.folio.circulation.domain.policy.Period;
+import org.folio.circulation.support.BadRequestFailure;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.CollectionResourceClient;
+import org.folio.circulation.support.RecordNotFoundFailure;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.ServerErrorFailure;
+import org.folio.circulation.support.http.client.Response;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.LocalDate;
+import org.joda.time.LocalTime;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import static api.support.fixtures.OpeningHourExamples.afterNoon;
+import static api.support.fixtures.OpeningHourExamples.allDay;
+import static api.support.fixtures.OpeningHourExamples.beforeNoon;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.folio.circulation.support.Result.failed;
+import static org.folio.circulation.support.Result.succeeded;
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.matches;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(JUnitParamsRunner.class)
+public class OverduePeriodCalculatorServiceTest {
```

Comment:
I find these kinds of tests difficult to follow.

Why set up all of the state needed for all of the tests together and then rely on parameter matching, over only setting up the state needed for each test within the test?

The former requires that the developer trying to understand the tests understand the complete set up and the logic for the matching. It effectively creates implicit branching within the tests.

Reply:
Test are now rewritten from scratch.

Reply:
Thank you for rewriting these tests to be more contained and less reliant on shared state.